pinot-core/src/test/java/com/linkedin/pinot/core/startree/hll/TestHllIndexSize.java

```diff
@@ -0,0 +1,187 @@
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.attribute.FileTime;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Dictionary Index Size for Hll Field is roughly 10 times of the corresponding index for Long field.
+ */
+public class TestHllIndexSize {
+  private static final Logger LOGGER = LoggerFactory.getLogger(TestHllIndexSize.class);
+  private static final String hllDeriveColumnSuffix = HllConstants.DEFAULT_HLL_DERIVE_COLUMN_SUFFIX;
+
+  // change this to change the columns that need to create hll index on
+  private static final List<String> columnsToDeriveHllFields =
+          Arrays.asList("column1", "column2", "column3",
+                  "count", "weeksSinceEpochSunday", "daysSinceEpoch",
+                  "column17", "column18");
+  private static final String AVRO_DATA = "data/test_data-sv.avro";
+  private static final String timeColumnName = "daysSinceEpoch";
+  private static final TimeUnit timeUnit = TimeUnit.DAYS;
+
+  private static final int hllLog2m = 5;
+
+  private IndexLoadingConfigMetadata v1LoadingConfig;
+  private IndexLoadingConfigMetadata v3LoadingConfig;
+
+  private HllConfig hllConfig;
+
+  @BeforeMethod
+  public void setUp() throws Exception {
+    hllConfig = new HllConfig(columnsToDeriveHllFields, hllLog2m, hllDeriveColumnSuffix);
+
+    Configuration tableConfig = new PropertiesConfiguration();
+    tableConfig.addProperty(IndexLoadingConfigMetadata.KEY_OF_SEGMENT_FORMAT_VERSION, "v1");
+    v1LoadingConfig = new IndexLoadingConfigMetadata(tableConfig);
+
+    tableConfig.clear();
+    tableConfig.addProperty(IndexLoadingConfigMetadata.KEY_OF_SEGMENT_FORMAT_VERSION,  "v3");
+    v3LoadingConfig = new IndexLoadingConfigMetadata(tableConfig);
+  }
+
+  @AfterMethod
+  public void tearDown() throws Exception {}
+
+  @Test
+  public void testColumnStatsWithoutStarTree() {
+    SegmentWithHllIndexCreateHelper helper = null;
+    boolean hasException = false;
+    try {
+      LOGGER.info("================ Without StarTree ================");
+      helper = new SegmentWithHllIndexCreateHelper(
+              "noStarTree", AVRO_DATA, timeColumnName, timeUnit);
+      SegmentIndexCreationDriver driver = helper.build(false, null);
+      LOGGER.info("================ Cardinality ================");
+      for (String name : helper.getSchema().getColumnNames()) {
+        LOGGER.info("* " + name + ": " + driver.getColumnStatisticsCollector(name).getCardinality());
+      }
+    } catch (Exception e) {
+      hasException = true;
+      LOGGER.error(e.getMessage());
+    } finally {
+      if (helper != null) {
+        helper.cleanTempDir();
+      }
+      Assert.assertEquals(hasException, false);
+    }
+  }
+
+  @Test
+  public void testColumnStatsWithStarTree() {
```

Comment:
It verifies the correctness of generating startree with or without hll and write to disk successfully without exceptions.